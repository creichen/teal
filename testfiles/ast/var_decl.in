// uninitilized variable declaration
a : MyType;
/*
	initialized variable declarations
 */
b : int := 0x40 ;

c : int := 10 ;

d : String := "escaped\" string" ;

// a fancy typed variable
e : QualId [typeActual] (funActual) QualId2 (funActual2) TypeId := 5;
