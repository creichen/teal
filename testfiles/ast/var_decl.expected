Program
  List
  List
    VarDecl
      IdDecl ID="a"
      Opt
        Type
          IdUse ID="MyType"
          List
          List
      Opt
    VarDecl
      IdDecl ID="b"
      Opt
        Type
          IdUse ID="int"
          List
          List
      Opt
        IntConstant Value="64"
    VarDecl
      IdDecl ID="c"
      Opt
        Type
          IdUse ID="int"
          List
          List
      Opt
        IntConstant Value="10"
    VarDecl
      IdDecl ID="d"
      Opt
        Type
          IdUse ID="String"
          List
          List
      Opt
        StringConstant Value="escaped\" string"
    VarDecl
      IdDecl ID="e"
      Opt
        Type
          IdUse ID="TypeId"
          List
          List
            TypeQualifier
              IdUse ID="QualId2"
              List
              List
                Access
                  IdUse ID="funActual2"
            TypeQualifier
              IdUse ID="QualId"
              List
                Type
                  IdUse ID="typeActual"
                  List
                  List
              List
                Access
                  IdUse ID="funActual"
      Opt
        IntConstant Value="5"
